### PR TITLE
Handle test notifications on iOS

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
Handle test notifications in iOS.

#### Related PRs
Server: https://github.com/mattermost/mattermost-server/pull/19610
Push Proxy: https://github.com/mattermost/mattermost-push-proxy/pull/96

#### Ticket Link
None

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: ios/Mattermost/AppDelegate.m
Comment (enahum) [inline ios/Mattermost/AppDelegate.m:94]:
```suggestion
  BOOL isTestAction = (action && [action isEqualToString: NOTIFICATION_TEST_ACTION]);
  
if (isTestAction) {
    completionHandler(UIBackgroundFetchResultNoData);
    return;
}
```

File: ios/Mattermost/AppDelegate.m
Comment (enahum) [inline ios/Mattermost/AppDelegate.m:105]:
```suggestion
```

**Review by enahum — CHANGES_REQUESTED** [review]
let's return early

File: ios/Mattermost/AppDelegate.m
Comment (enahum) [inline ios/Mattermost/AppDelegate.m:96]: hahahaha weird indentation

**Review by enahum — APPROVED** [review]

**Comment by larkox** [comment]
@enahum Just to confirm, do we want to merge this already? (I don't see any reason to delay merging)

**Comment by enahum** [comment]
I did already approved, 0/5 if you want to merge without another approval

**Review by avinashlng1080 — APPROVED** [review]

**Pull request merged**
